ngraph_bridge/ie_executable.cc

```diff
@@ -40,9 +40,46 @@ IE_Executable::IE_Executable(shared_ptr<Function> func, string device)
     }
   }
 
+  NGRAPH_VLOG(2) << "Checking for function parameters in IE backend";
+  if (func->get_parameters().size() == 0) {
+    NGRAPH_VLOG(1) << "No parameters found in nGraph function!";
+    // Try to find a node that can be converted into a "static input"
+    bool param_replaced = false;
+    for (const auto& node : func->get_ordered_ops()) {
+      // Only try to convert constant nodes at the edge to parameters
+      // FIXME: IE cannot handle input parameters with i64/u6 precision
+      // at the moment
+      if (node->get_input_size() == 0 && node->is_constant() &&
+          !(node->get_element_type() == ngraph::element::i64 ||
+            node->get_element_type() == ngraph::element::u64)) {
+        auto constant = ngraph::as_type_ptr<opset::Constant>(node);
+        auto element_type = constant->get_element_type();
+        auto shape = constant->get_shape();
+        auto param = std::make_shared<opset::Parameter>(element_type, shape);
+        ngraph::replace_node(node, param);
+        // nGraph doesn't provide a way to set a parameter to an existing
+        // function, so we clone the function here...
+        func = make_shared<Function>(func->get_results(),
+                                     ParameterVector{param}, func->get_name());
+        auto ie_tensor = make_shared<IETensor>(element_type, shape);
+        ie_tensor->write(constant->get_data_ptr(),
+                         shape_size(shape) * element_type.size());
+        m_hoisted_params[param->get_name()] = ie_tensor;
+        NGRAPH_VLOG(1) << "Converted node " << constant << " to a parameter "
+                       << param;
+        param_replaced = true;
+        break;
+      }
+      if (!param_replaced) {
+        THROW_IE_EXCEPTION << "Found a function with no parameters!";
```

Comment:
May be change the error message here to say that "Unable to add a param to a func with no params"

Reply:
Done.